src/app/OperationalSessionSetup.cpp

```diff
@@ -242,6 +244,8 @@ void OperationalSessionSetup::UpdateDeviceData(const Transport::PeerAddress & ad
         return;
     }
 
+    MATTER_LOG_METRIC_END(kMetricDeviceOperationalDiscovery, CHIP_NO_ERROR);
```

Comment:
So how will this work in practice when we get multiple UpdateDeviceData for a single discovery start (e.g. if we found multiple IPs and the first one we tried did not work)?